src/neo/Network/P2P/ProtocolHandler.cs

```diff
@@ -47,8 +47,8 @@ public ProtocolHandler(NeoSystem system)
         {
             this.system = system;
             this.pendingKnownHashes = new PendingKnownHashesCollection();
-            this.knownHashes = new FIFOSet<UInt256>(Blockchain.Singleton.MemPool.Capacity * 2);
-            this.sentHashes = new FIFOSet<UInt256>(Blockchain.Singleton.MemPool.Capacity * 2);
+            this.knownHashes = new HashSetCache<UInt256>(Blockchain.Singleton.MemPool.Capacity * 2 / 10);
+            this.sentHashes = new HashSetCache<UInt256>(Blockchain.Singleton.MemPool.Capacity * 2 / 10);
```

Comment:
Why / 10?

Reply:
Because input parameter is each bucket capacity not the total capacity.